### PR TITLE
Created documentation for command hooks feature

### PR DESCRIPTION
**Pull request by renebrandel**
_Issue #, if available:_

_Description of changes:_
- Added documentation for upcoming command hooks feature
- Using this PR in favor of #3486 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/cli/usage/menu.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "permissions-boundary",
     "headless",
     "upgrade",
-    "uninstall"
+    "uninstall",
+    "runtime-hooks"
```
Comment (akshbhu) [inline docs/cli/usage/menu.json:16]: command

File: src/pages/cli/usage/command-hooks.mdx [hunk 2]
```diff
@@ -0,0 +1,250 @@
+export const meta = {
+  title: `Command Hooks`,
+  description: `Execute custom scripts before, during, and after Amplify CLI commands (“amplify push”, “amplify api gql-compile”, and more). This allows you to extend Amplify’s best-practice defaults to meet your organization’s specific security guidelines and operational requirements.`,
```
Comment (edwardfoyle) [inline src/pages/cli/usage/command-hooks.mdx:3]: This seems a little verbose for the page description. I'd suggest just the first sentence here

File: src/pages/cli/usage/command-hooks.mdx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+### Managing third-party dependencies
+
+Packages from external package managers like `npm` can be used in command hooks scripts. In this example, we'll showcase how to install `axios` as a dependency for our Node.js hooks.
+
+First, go to the hooks folder and install the `axios` dependency.
+```
+> cd amplify/hooks
+> npm init
+> npm install axios
```
Comment (akshbhu) [inline src/pages/cli/usage/command-hooks.mdx:194]: We also need to have a prebuild step in console to install hooks dependencies

```
backend:
        phases:
            preBuild:
            commands:
                - cd amplify/hooks
                - npm install axios
        frontend:
```

File: src/pages/cli/usage/command-hooks.mdx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+};
+
+Use Command Hooks to execute custom scripts before, during, and after Amplify CLI commands (“amplify push”, “amplify api gql-compile”, and more). This allows you to extend Amplify’s best-practice defaults to meet your organization’s specific security guidelines and operational requirements. 
+
+## Adding a command hook
+Place your custom scripts in the `amplify/hooks` directory and set the script file name to the desired command with a pre or post designation. For example `post-add-function.js` will execute the script after `amplify add function`. For more information about the script naming convention, see [How to name command hook scripts](#how-to-name-command-hook-scripts). 
+
+For this example, let's create a hook to ensure that a minimum major Amplify CLI version is used before deployment (`amplify push`).
+
+Let's add `pre-push.js` in the `amplify/hooks` drectory with the following contents. *Note: You need to create a `amplify/hooks` folder if your Amplify project was created prior to Amplify CLI version XYZ*
```
Comment (edwardfoyle) [inline src/pages/cli/usage/command-hooks.mdx:13]: replace XYZ with 5.5.0

File: src/pages/cli/usage/command-hooks.mdx [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+const parameters = JSON.parse(fs.readFileSync(0, { encoding: "utf8" }));
+
+// Get the running Amplify CLI major verison number
+const currentCLIMajorVersion = parameters.data.amplify.version.split('.')[0]
+console.log('Amplify CLI major version: ', currentCLIMajorVersion)
+
+const MINIMUM_MAJOR_AMPLIFY_CLI_VERSION = 5
+console.log('Minimum required Amplify CLI major version: ', MINIMUM_MAJOR_AMPLIFY_CLI_VERSION)
+
+if (currentCLIMajorVersion >= MINIMUM_MAJOR_AMPLIFY_CLI_VERSION) {
```
Comment (edwardfoyle) [inline src/pages/cli/usage/command-hooks.mdx:26]: shouldn't this be `<`?

File: src/pages/cli/usage/command-hooks.mdx [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+`status`                  |`notifications`<br/>
+`import`                  |`auth`, `storage`, `env`<br/>
+`gqlcompile`              |`api`<br/>
+`addgraphqldatasource`    |`api`<br/>
+`statements`              |`codegen`<br/>
+`types`                  |`codegen`<br/>
+
+
+
+*Note: Amplify CLI will throw error if multiple hook scripts with same filename exist in the hooks directory.*
```
Comment (edwardfoyle) [inline src/pages/cli/usage/command-hooks.mdx:84]:
```suggestion
*Note: Multiple hook scripts with the same filename are not allowed*
```

File: src/pages/cli/usage/command-hooks.mdx [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+```javascript
+const fs = require("fs");
+const parameters = JSON.parse(fs.readFileSync(0, { encoding: "utf8" }));
+console.log(parameters.data, parameters.error)
+```
+
+### How to access command hook parameters in Bash
+```bash
+parameters=`cat`
+data=$(jq -r '.data' <<< "$parameters")
```
Comment (edwardfoyle) [inline src/pages/cli/usage/command-hooks.mdx:132]: maybe add a note about installing `jq`?

File: src/pages/cli/usage/command-hooks.mdx [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+exit 1
+```
+
+</Block>
+
+</BlockSwitcher>
+
+
+## Advanced command hook configurations
+You can optionally add `hooks-config.json` file in the hooks directory to configure custom scripting runtimes or manage external dependencies.
```
Comment (edwardfoyle) [inline src/pages/cli/usage/command-hooks.mdx:163]:
```suggestion
You can optionally add `hooks-config.json` in `amplify/hooks` to configure custom scripting runtimes or manage external dependencies.
```

File: src/pages/cli/usage/command-hooks.mdx [hunk 9]
```diff
@@ -0,0 +1,250 @@
+export const meta = {
+  title: `Command Hooks`,
+  description: `Execute custom scripts before, during, and after Amplify CLI commands (“amplify push”, “amplify api gql-compile”, and more). This allows you to extend Amplify’s best-practice defaults to meet your organization’s specific security guidelines and operational requirements.`,
+};
+
+Use Command Hooks to execute custom scripts before, during, and after Amplify CLI commands (“amplify push”, “amplify api gql-compile”, and more). This allows you to extend Amplify’s best-practice defaults to meet your organization’s specific security guidelines and operational requirements. 
+
+## Adding a command hook
+Place your custom scripts in the `amplify/hooks` directory and set the script file name to the desired command with a pre or post designation. For example `post-add-function.js` will execute the script after `amplify add function`. For more information about the script naming convention, see [How to name command hook scripts](#how-to-name-command-hook-scripts).
```
Comment (akshbhu) [inline src/pages/cli/usage/command-hooks.mdx:9]: Can we also mention , hooks folder will get created when running amplify init with sample hooks

File: src/pages/cli/usage/command-hooks.mdx [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+</Block>
+
+</BlockSwitcher>
+
+
+## Advanced command hook configurations
+You can optionally add `hooks-config.json` file in the hooks directory to configure custom scripting runtimes or manage external dependencies.
+
+### Adding a custom scripting runtime 
+By default, Node.js and Bash are supported. To add more programming languages support, add `extensions` to a `hooks-config.json` file in the `amplify/hooks` folder.
```
Comment (edwardfoyle) [inline src/pages/cli/usage/command-hooks.mdx:166]:
```suggestion
By default, Node.js and Bash are supported. To support additional runtimes, add `extensions` to the `hooks-config.json` file in the `amplify/hooks` folder.
```

**Review by akshbhu — COMMENTED** [review]
2 nits , otherwise looks good

**Review by jhockett — APPROVED** [review]

File: src/pages/cli/usage/command-hooks.mdx (hunk 9 above)
Comment (renebrandel) [reply src/pages/cli/usage/command-hooks.mdx:9]: Already mentioned below.

File: src/pages/cli/usage/command-hooks.mdx (hunk 4 above)
Comment (renebrandel) [inline src/pages/cli/usage/command-hooks.mdx:13]:
```suggestion
Let's add `pre-push.js` in the `amplify/hooks` drectory with the following contents. *Note: You need to create a `amplify/hooks` folder if your Amplify project was created prior to Amplify CLI version 5.5.0*
```

File: src/pages/cli/usage/command-hooks.mdx (hunk 5 above)
Comment (renebrandel) [inline src/pages/cli/usage/command-hooks.mdx:26]:
```suggestion
if (currentCLIMajorVersion < MINIMUM_MAJOR_AMPLIFY_CLI_VERSION) {
```

File: src/pages/cli/usage/command-hooks.mdx [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+- stack - the error stack emitted by Amplify CLI
+
+### How to access command hook parameters in Node.js
+```javascript
+const fs = require("fs");
+const parameters = JSON.parse(fs.readFileSync(0, { encoding: "utf8" }));
+console.log(parameters.data, parameters.error)
+```
+
+### How to access command hook parameters in Bash
```
Comment (renebrandel) [inline src/pages/cli/usage/command-hooks.mdx:129]:
```suggestion
### How to access command hook parameters in Bash

First, install a JSON parser like [`jq`](https://stedolan.github.io/jq/download/).

Then, parse the parameters:
```

File: src/pages/cli/usage/command-hooks.mdx (hunk 2 above)
Comment (renebrandel) [reply src/pages/cli/usage/command-hooks.mdx:3]: Keeping as is for SEO purposes. 
```suggestion
  description: `Execute custom scripts before, during, and after Amplify CLI commands (“amplify push”, “amplify api gql-compile”, and more).`,
```

File: src/pages/cli/usage/command-hooks.mdx [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+## Adding a command hook
+Place your custom scripts in the `amplify/hooks` directory and set the script file name to the desired command with a pre or post designation. For example `post-add-function.js` will execute the script after `amplify add function`. For more information about the script naming convention, see [How to name command hook scripts](#how-to-name-command-hook-scripts). 
+
+For this example, let's create a hook to ensure that a minimum major Amplify CLI version is used before deployment (`amplify push`).
+
+Let's add `pre-push.js` in the `amplify/hooks` drectory with the following contents. *Note: You need to create a `amplify/hooks` folder if your Amplify project was created prior to Amplify CLI version 5.5.0*
+
+```js
+const fs = require("fs");
+const parameters = JSON.parse(fs.readFileSync(0, { encoding: "utf8" }));
```
Comment (jhockett) [inline src/pages/cli/usage/command-hooks.mdx:17]: nit: use consistent quotation

```suggestion
const fs = require('fs');
const parameters = JSON.parse(fs.readFileSync(0, { encoding: 'utf8' }));
```

File: src/pages/cli/usage/command-hooks.mdx [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+  "stack": String
+}
+```
+- message - the error message emitted by Amplify CLI
+- stack - the error stack emitted by Amplify CLI
+
+### How to access command hook parameters in Node.js
+```javascript
+const fs = require("fs");
+const parameters = JSON.parse(fs.readFileSync(0, { encoding: "utf8" }));
```
Comment (jhockett) [inline src/pages/cli/usage/command-hooks.mdx:125]:
```suggestion
const fs = require('fs');
const parameters = JSON.parse(fs.readFileSync(0, { encoding: 'utf8' }));
```

**Review by jhockett — APPROVED** [review]

**Review by akshbhu — APPROVED** [review]
LGTM

**Pull request merged**